corelib-definitions/src/main/java/eu/europeana/corelib/definitions/solr/model/Query.java

```diff
@@ -251,20 +255,106 @@ public Query setFacets(List<String> facets) {
+                 StringUtils.equalsIgnoreCase("IMAGE_COLOR", facet)) {
+          ok = true;
+        }
+        else if (StringUtils.equalsIgnoreCase("IMAGE_GREYSCALE", facet) ||
+                 StringUtils.equalsIgnoreCase("IMAGE_GRAYSCALE", facet)) {
+          ok = true;
+          final int tag = MediaTypeEncoding.IMAGE.getEncodedValue();
+          generateFacetTagQuery (tag | (1 << TagEncoding.IMAGE_COLOURSPACE.getBitPos()),
+                                 tag | (2 << TagEncoding.IMAGE_COLOURSPACE.getBitPos()),
+                                 tag | (3 << TagEncoding.IMAGE_COLOURSPACE.getBitPos())
+                                );
+        }
+        else if (StringUtils.equalsIgnoreCase("IMAGE_ASPECTRATIO", facet)) {
+          ok = true;
+          final int tag = MediaTypeEncoding.IMAGE.getEncodedValue();
+          generateFacetTagQuery (tag | (1 << TagEncoding.IMAGE_ASPECTRATIO.getBitPos()),
+                                 tag | (2 << TagEncoding.IMAGE_ASPECTRATIO.getBitPos())
+                                );
+        }
+        else if (StringUtils.equalsIgnoreCase("VIODE_HD", facet)) {
+          ok = true;
+            final int tag = MediaTypeEncoding.VIDEO.getEncodedValue();
+            generateFacetTagQuery (tag | (1 << TagEncoding.VIDEO_QUALITY.getBitPos()),
+                                   tag | (0 << TagEncoding.VIDEO_QUALITY.getBitPos())
+                                  );
+        }
+        else if (StringUtils.equalsIgnoreCase("VIDEO_DURATION", facet)) {
+          ok = true;
+            final int tag = MediaTypeEncoding.VIDEO.getEncodedValue();
+            generateFacetTagQuery (tag | (1 << TagEncoding.VIDEO_DURATION.getBitPos()),
+                                   tag | (2 << TagEncoding.VIDEO_DURATION.getBitPos()),
+                                   tag | (3 << TagEncoding.VIDEO_DURATION.getBitPos())
+                                  );
+        }
+        else if (StringUtils.equalsIgnoreCase("SOUND_DURATION", facet)) {
+          ok = true;
+          final int tag = MediaTypeEncoding.SOUND.getEncodedValue();
+          generateFacetTagQuery (tag | (1 << TagEncoding.SOUND_DURATION.getBitPos()),
+                                 tag | (0 << TagEncoding.SOUND_DURATION.getBitPos())
+                                 );
+        }
+        else if (StringUtils.equalsIgnoreCase("SOUND_HQ", facet)) {
+          ok = true;
+            final int tag = MediaTypeEncoding.SOUND.getEncodedValue();
+            generateFacetTagQuery (tag | (1 << TagEncoding.SOUND_DURATION.getBitPos()),
+                                   tag | (2 << TagEncoding.SOUND_DURATION.getBitPos())
+                                  );
+        }
+        */
+        else additionalFacets.add(facet);
+    }
+		  facets = additionalFacets;
 	}
 
-	public boolean isApiQuery() {
+    private void generateFacetTagQuery (int ... tags) {
+        if (null == tags || 0 == tags.length) {
```

Comment:
It's not an easy to use a for each on an empty not null array so
if (tags != null) {
for (int tag: tags) {
...
}
}